src/bin/bat/config.rs

```diff
@@ -15,6 +16,46 @@ pub fn config_file() -> PathBuf {
         .unwrap_or_else(|| PROJECT_DIRS.config_dir().join("config"))
 }
 
+pub fn generate_config_file() {
+    let config_file = config_file();
+    if config_file.exists() {
+        println!("A config file already exists at: {}", config_file.to_string_lossy());
+
+        print!("Overwrite? (y/n): ");
+        let _ = io::stdout().flush();
+        let mut decision = String::new();
+        io::stdin().read_line(&mut decision).expect("Failed to read input");
```

Comment:
If this function returns a `Result<…>`, you can simply use `.read_line(&mut decision)?` instead of `.expect(…)` which would kill the process in case of an error.